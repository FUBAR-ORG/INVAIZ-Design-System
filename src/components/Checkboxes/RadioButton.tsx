import type { CheckboxDefaultProps } from '@components/Checkboxes/interfaces/Checkbox.interface';
// types

import HoverCircle from '@components/Checkboxes/HoverCircle';
// components

import {
  StyleRadioButton,
  StyleRadioButtonInput,
} from '@components/Checkboxes/styles/RadioButton.style';
// styles

/**
 * 라디오 버튼을 나타내는 컴포넌트입니다.
 */
const RadioButton = ({ disabled, onChange, ...props }: CheckboxDefaultProps) => (
  <HoverCircle disabled={disabled}>
    <StyleRadioButtonInput
      type='checkbox'
      onChange={(e) => onChange?.(e.target.checked)}
      disabled={disabled}
      {...props}
    />
    <StyleRadioButton />
  </HoverCircle>
);

export default RadioButton;
