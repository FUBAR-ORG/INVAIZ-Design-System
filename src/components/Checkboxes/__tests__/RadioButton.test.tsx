import RadioButton from '@components/Checkboxes/RadioButton';
import { render } from '@tests/test-utils';

describe('RadioButton', () => {
  it('라디오 버튼의 기본값은 `false`이며, 변경이 가능하다.', () => {
    const { getAllByRole } = render(
      <>
        <RadioButton />
        <RadioButton checked />
      </>
    );
    const [isNotChanged, isChanged] = getAllByRole('checkbox');

    expect(isNotChanged).not.toBeChecked();
    expect(isChanged).toBeChecked();
  });

  it('라디오 버튼은 클릭할 수 있으며, 클릭 시 `checked` 상태 변경이 가능하다.', () => {
    const onChange = jest.fn();
    const { getByRole } = render(<RadioButton onChange={onChange} />);
    const checkbox = getByRole('checkbox');

    expect(onChange).not.toHaveBeenCalled();

    checkbox.click();
    expect(onChange).toHaveBeenCalledWith(true);
    expect(checkbox).toBeChecked();

    checkbox.click();
    expect(onChange).toHaveBeenCalledWith(false);
    expect(checkbox).not.toBeChecked();

    expect(onChange).toHaveBeenCalled();
  });

  it('라디오 버튼이 `disabled`일 때, 상태 변경이 불가능하다.', () => {
    const { getByRole } = render(<RadioButton disabled />);
    const checkbox = getByRole('checkbox');

    checkbox.click();
    expect(checkbox).not.toBeChecked();
  });
});
