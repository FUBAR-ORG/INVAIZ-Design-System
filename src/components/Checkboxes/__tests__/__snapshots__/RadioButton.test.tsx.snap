// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Snapshot 1`] = `
<span
  className="css-srlyvc"
>
  <input
    checked={true}
    className="css-1nwbmtb"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="css-1umxdg8"
  />
</span>
`;

exports[`RadioButton Snapshot 2`] = `
<span
  className="css-srlyvc"
>
  <input
    className="css-1nwbmtb"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="css-1umxdg8"
  />
</span>
`;
